core/server/api/shared/http.js

```diff
@@ -15,7 +15,8 @@ const http = (apiImpl) => {
             user: req.user,
             context: {
                 api_key_id: (req.api_key && req.api_key.id) ? req.api_key.id : null,
-                user: ((req.user && req.user.id) || (req.user && models.User.isExternalUser(req.user.id))) ? req.user.id : null
+                user: ((req.user && req.user.id) || (req.user && models.User.isExternalUser(req.user.id))) ? req.user.id : null,
```

Comment:
I know it's not related to this PR, but I feel like I'm missing something here - if I'm following the logic correctly, the secondary boolean should always be falsy since `req.user.id` won't exist?

Reply:
:grin: good eyes! This is a weird check thought - the first check `req.user && req.user.id` checks if the id is truthy. so we store it.

If that fails than either the `id` is falsy, but not necessarily `undefined`.

The `module.User.isExternalUser` is essentially a check for `id === 0` - in which case we want to set that, and otherwise null.

That was wordy, did it make sense?